### PR TITLE
Rework SDK

### PR DESCRIPTION
**Pull request by willtraweek**

### CODE REVIEW

File: codelighthouse/CodeLighthouse.py [hunk 1]
```diff
@@ -4,23 +4,63 @@
...
+        for key, value in kwargs.items():
+            output[key] = value
+
+        return output
+
+    @staticmethod
+    def format_stack_trace(trace):
+        # traceback allows for locals to be found up and down the stack.  Call trace.f_locals for them
+        # print(trace)
+        return
```
Comment (K-Mistele) [inline codelighthouse/CodeLighthouse.py]: Shouldn't there be code here?

File: codelighthouse/CodeLighthouse.py [hunk 2]
```diff
@@ -4,23 +4,63 @@
...
+        self.web_handler.x_api_key = x_api_key
+
+        if environment == "local":
+            self.web_handler.BASE_URL = "http://localhost:5000"
+            self.web_handler.DEBUG = True
+        elif environment == "dev":
+            self.web_handler.BASE_URL = "https://dev.codelighthouse.io"
+            self.web_handler.DEBUG = True
+        else:
+            self.web_handler.BASE_URL = "https://codelighthouse.io"
```
Comment (K-Mistele) [inline codelighthouse/CodeLighthouse.py:27]: Wouldn't this possibly pick up the client's environment variables? For example, if they're in their dev environment and have an environment variable set to "dev", they're gonna send the request to `dev.codelighthouse.io` which isn't what we want.

**Review by K-Mistele — CHANGES_REQUESTED** [review]

**Comment by K-Mistele** [comment]
Couple additional thoughts on this: 
1. we probably don't need them to specify an organization name, right? It'll be tied to their API Key and can be looked up with the API key. If we do need them to specify it, it probably doesn't need to be done with an environment variable.
2. per my comment in the other PR, I'd like to see resource group and resource name being included as well
3. For configuring the module, what I'm envisioning is something like this: they create a config disctionary or whatever and then pass it to the class:

```
options = {
"x_api_key": os.environ.get("CODELIGHTHOUSE_SECRET"),
"environment": os.environ.get("ENVIRONMENT"),
"resource_group": "AWS-lambdas",
"resource_name": "example-lambda"
}
lighthouse = CodeLighthouse(options)
```

File: codelighthouse/CodeLighthouse.py (hunk 2 above)
Comment (willtraweek) [reply codelighthouse/CodeLighthouse.py:27]: environment is manually set.  If someone is dumb enough to manually set it to "local" then thats their problem

**Review by K-Mistele — APPROVED** [review]
looks good

**Pull request merged**